File: network/src/test/java/quan/network/RoleInfo.java
package quan.network;

import quan.network.message.Bean;
import quan.network.message.Buffer;

import java.io.IOException;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.HashSet;

public class RoleInfo extends Bean {

    private long roleId;//角色id
    private boolean bo;
    private byte by;
    private short s;
    private int i;
    private float f;
    private double d;
    private String roleName;//角色名
    private byte[] data;
    private ArrayList<Integer> list;
    private HashSet<Integer> set;
    private HashMap<Integer, Integer> map;

    public RoleInfo() {
        roleId = 111L;
        bo = true;
        by = (byte) 22;
        s = (short) 22;
        i = 11;
        f = 22.33332F;
        d = 33.332432D;
        roleName = "zhangsan";
        data = new byte[0];
        list = new ArrayList<>();
        set = new HashSet<>();
        map = new HashMap<>();
    }

    public long getRoleId() {
        return roleId;
    }

    public void setRoleId(long roleId) {
        this.roleId = roleId;
    }

    public boolean getBo() {
        return bo;
    }

    public void setBo(boolean bo) {
        this.bo = bo;
    }

    public byte getBy() {
        return by;
    }

    public void setBy(byte by) {
        this.by = by;
    }

    public short getS() {
        return s;
    }

    public void setS(short s) {
        this.s = s;
    }

    public int getI() {
        return i;
    }

    public void setI(int i) {
        this.i = i;
    }

    public float getF() {
        return f;
    }

    public void setF(float f) {
        this.f = f;
    }

    public double getD() {
        return d;
    }

    public void setD(double d) {
        this.d = d;
    }

    public String getRoleName() {
        return roleName;
    }

    public void setRoleName(String roleName) {
        this.roleName = roleName;
    }


    public byte[] getData() {
        return data;
    }

    public void setData(byte[] data) {
        this.data = data;
    }

    public ArrayList<Integer> getList() {
        return list;
    }

    public HashSet<Integer> getSet() {
        return set;
    }

    public HashMap<Integer, Integer> getMap() {
        return map;
    }


    @Override
    public void encode(Buffer buffer) throws IOException {
        buffer.writeLong(roleId);
        buffer.writeBool(bo);
        buffer.writeByte(by);
        buffer.writeShort(s);
        buffer.writeInt(i);
        buffer.writeFloat(f);
        buffer.writeDouble(d);
        buffer.writeString(roleName);
        buffer.writeBytes(data);
        buffer.writeInt(list.size());
        for (int listValue : list) {
            buffer.writeInt(listValue);
        }
        buffer.writeInt(set.size());
        for (int setValue : set) {
            buffer.writeInt(setValue);
        }
        buffer.writeInt(map.size());
        for (int mapKey : map.keySet()) {
            buffer.writeInt(mapKey);
            buffer.writeInt(map.get(mapKey));
        }
    }

    @Override
    public void decode(Buffer buffer) throws IOException {
        roleId = buffer.readLong();
        bo = buffer.readBool();
        by = buffer.readByte();
        s = buffer.readShort();
        i = buffer.readInt();
        f = buffer.readFloat();
        d = buffer.readDouble();
        roleName = buffer.readString();
        data = buffer.readBytes();
        int listSize = buffer.readInt();
        for (int i = 0; i < listSize; i++) {
            list.add(buffer.readInt());
        }
        int setSize = buffer.readInt();
        for (int i = 0; i < setSize; i++) {
            set.add(buffer.readInt());
        }
        int mapSize = buffer.readInt();
        for (int i = 0; i < mapSize; i++) {
            map.put(buffer.readInt(), buffer.readInt());
        }
    }

    @Override
    public String toString() {
        return "RoleInfo{" +
                "roleId=" + roleId +
                ",bo=" + bo +
                ",by=" + by +
                ",s=" + s +
                ",i=" + i +
                ",f=" + f +
                ",d=" + d +
                ",roleName='" + roleName + '\'' +
                ",data=" + Arrays.toString(data) +
                ",list=" + list +
                ",set=" + set +
                ",map=" + map +
                '}';

    }

}

File: quan-cs/Test/Message/Role/RoleInfo.cs
using System;
using System.Collections.Generic;
using Quan.Common;
using Quan.Message;
using Buffer = Quan.Message.Buffer;

namespace Test.Message.Role
{
	/// <summary>
	/// 角色信息<br/>
	/// 自动生成
	/// </summary>
    public class RoleInfo : Bean
    {
        /// <summary>
		/// 角色id
		/// </summary>
		public long Id { get; set; }

		private string _roleName = "";

        /// <summary>
		/// 角色名
		/// </summary>
		public string RoleName
		{
	    	get => _roleName;
	    	set => _roleName = value ?? throw new NullReferenceException();
		}

		public RoleType RoleType { get; set; }

		public bool B { get; set; }

		public short S { get; set; }

		public int I { get; set; }

		public float F { get; set; }

		public double D { get; set; }

		private byte[] _data = new byte[0];

		public byte[] Data
		{
            get => _data;
            set => _data = value ?? throw new NullReferenceException();
		}

		public List<int> List { get; } = new List<int>();

		public HashSet<int> Set { get; } = new HashSet<int>();

		public Dictionary<int, int> Map { get; } = new Dictionary<int, int>();



		public override void Encode(Buffer buffer)
		{
	    	base.Encode(buffer);

		    buffer.WriteLong(Id);
		    buffer.WriteString(RoleName);
			buffer.WriteInt((int)RoleType);
		    buffer.WriteBool(B);
		    buffer.WriteShort(S);
		    buffer.WriteInt(I);
		    buffer.WriteFloat(F);
		    buffer.WriteDouble(D);
		    buffer.WriteBytes(Data);

		    buffer.WriteInt(List.Count);
		    foreach (var listValue in List) {
			    buffer.WriteInt(listValue);
		    }

		    buffer.WriteInt(Set.Count);
		    foreach (var setValue in Set) {
			    buffer.WriteInt(setValue);
		    }

		    buffer.WriteInt(Map.Count);
		    foreach (var mapKey in Map.Keys) {
		        buffer.WriteInt(mapKey);
			    buffer.WriteInt(Map[mapKey]);
		    }
		}

		public override void Decode(Buffer buffer)
		{
	    	base.Decode(buffer);

		    Id = buffer.ReadLong();
		    RoleName = buffer.ReadString();
		    RoleType = (RoleType)buffer.ReadInt();
		    B = buffer.ReadBool();
		    S = buffer.ReadShort();
		    I = buffer.ReadInt();
		    F = buffer.ReadFloat();
		    D = buffer.ReadDouble();
		    Data = buffer.ReadBytes();

		    var listSize = buffer.ReadInt();
		    for (var i = 0; i < listSize; i++) {
			    List.Add(buffer.ReadInt());
		    }

		    var setSize = buffer.ReadInt();
		    for (var i = 0; i < setSize; i++) {
			    Set.Add(buffer.ReadInt());
		    }

		    var mapSize = buffer.ReadInt();
		    for (var i = 0; i < mapSize; i++) {
			    Map.Add(buffer.ReadInt(), buffer.ReadInt());
		    }
		}

		public override string ToString()
		{
			return "RoleInfo{" +
					"id=" + Id.ToString2() +
					",roleName='" + RoleName + '\'' +
					",roleType=" + RoleType.ToString2() +
					",b=" + B.ToString2() +
					",s=" + S.ToString2() +
					",i=" + I.ToString2() +
					",f=" + F.ToString2() +
					",d=" + D.ToString2() +
					",data=" + Data.ToString2() +
					",list=" + List.ToString2() +
					",set=" + Set.ToString2() +
					",map=" + Map.ToString2() +
					'}';
		}
    }
}